server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1178,7 +1182,7 @@ public Map<String, String> getCustomData(final String key) {
         return this.customData.get(key);
     }
 
-    public Map<Integer, Set<String>> getInSyncAllocationIds() {
+    public List<Set<String>> getInSyncAllocationIds() {
         return inSyncAllocationIds;
```

Comment:
It looks like this exposes mutable collection. Is there a chance we could make it immutable after initialization?

Reply:
@idegtiarenko I am not sure if it is possible, but I can check. I will try it out with using `Collections.unmodifiableList()` -- if that's not what you had in mind, please tell me.

Reply:
Made it so that during the `IndexMetadata` constructor, it makes the list unmodifiable. Hope this resolves this. If not, please unresolve the conversation and tell me.